client/data/settings/hooks.js

```diff
@@ -0,0 +1,64 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from '../constants';
+
+export const useSettings = () => {
+	const { saveSettings } = useDispatch( STORE_NAME );
+
+	return useSelect(
+		( select ) => {
+			const {
+				getSettings,
+				hasFinishedResolution,
+				isResolving,
+				isSavingSettings,
+			} = select( STORE_NAME );
+
+			const isLoading =
+				isResolving( 'getSettings' ) ||
+				! hasFinishedResolution( 'getSettings' );
+
+			return {
+				settings: getSettings(),
+				isLoading,
+				saveSettings,
+				isSaving: isSavingSettings(),
+			};
+		},
+		[ saveSettings ]
+	);
+};
+
+export const usePaymentRequestEnabledSettings = () => {
+	const { updateIsPaymentRequestEnabled } = useDispatch( STORE_NAME );
+
+	return useSelect( ( select ) => {
+		const { getIsPaymentRequestEnabled } = select( STORE_NAME );
+
+		return [ getIsPaymentRequestEnabled(), updateIsPaymentRequestEnabled ];
+	} );
```

Comment:
I recommend separating the value returned by the selector from the value returned by the hook.
Otherwise the selector is returning a new array reference (which it doesn't need to do) and it makes harder to memoize the returned value.
The `useSelect` doesn't need to know about a change on `updateIsPaymentRequestEnabled`, this way.

Just like in `react-redux`, it looks like `useSelect` "remembers" the value returned by the callback function based on the dependencies. So if the dependencies didn't change and the store didn't change, it can just return a "cached" value: https://github.com/WordPress/gutenberg/blob/f242c91b21bbbccf0a0129e4946aadfa4d04baca/packages/data/src/components/use-select/index.js#L109

```suggestion
	const isPaymentRequestEnabled = useSelect( ( select ) => {
		const { getIsPaymentRequestEnabled } = select( STORE_NAME );

		return getIsPaymentRequestEnabled();
	}, [] );

	return [ isPaymentRequestEnabled, updateIsPaymentRequestEnabled ];
```

Reply:
Fantastic! :rocket: